packages/rest/src/rest.server.ts

```diff
@@ -272,14 +290,24 @@ export class RestServer extends Context implements Server, HttpServerLike {
     const mapping = this.config.openApiSpec!.endpointMapping!;
     // Serving OpenAPI spec
     for (const p in mapping) {
-      this._expressApp.get(p, (req, res) =>
-        this._serveOpenApiSpec(req, res, mapping[p]),
+      expressCtx.middleware(
+        (req, res) => this._serveOpenApiSpec(req, res, mapping[p]),
+        {
+          path: p,
+          method: 'get',
+          phase: 'openapi-spec',
+        },
       );
     }
 
     const explorerPaths = ['/swagger-ui', '/explorer'];
-    this._expressApp.get(explorerPaths, (req, res, next) =>
-      this._redirectToSwaggerUI(req, res, next),
+    expressCtx.middleware(
+      (req, res, next) => this._redirectToSwaggerUI(req, res, next),
```

Comment:
Similarly here. We don't support dynamic redirects yet - see https://github.com/strongloop/loopback-next/issues/2022 and the work in progress in https://github.com/strongloop/loopback-next/pull/2529, but I think it should be possible to implement "redirecto to Swagger UI" by implementing a LB4 route that's calling `response.redirect` directly.

Reply:
I think this change should be deferred until #2529 and #2547 are landed, at which point we should rework these redirects to leverage `server.redirect` APIs instead of calling `response.redirect` directly.